src/utilities/BatchTableHierarchyExtension.js

```diff
@@ -0,0 +1,122 @@
+import { parseBinArray } from './FeatureTable.js';
+
+export class BatchTableHierarchyExtension {
+
+	constructor( batchTable ) {
+
+		this.batchTable = batchTable;
+
+		const extensionHeader = batchTable.header.extensions[ '3DTILES_batch_table_hierarchy' ];
+
+		this.classes = extensionHeader.classes;
+		for ( const classDef of this.classes ) {
+
+			const instances = classDef.instances;
+			for ( const property in instances ) {
+
+				classDef.instances[ property ] = this._parseProperty( instances[ property ], classDef.length );
+
+			}
+
+		}
+
+		this.instancesLength = extensionHeader.instancesLength;
+
+		this.classIds = this._parseProperty( extensionHeader.classIds, this.instancesLength );
+
+		if ( extensionHeader.parentCounts ) {
+
+			this.parentCounts = this._parseProperty( extensionHeader.parentCounts, this.instancesLength );
+
+		} else {
+
+			this.parentCounts = new Array( this.instancesLength ).fill( 1 );
+
+		}
+
+		if ( extensionHeader.parentIds ) {
+
+			this.parentIds = this._parseProperty( extensionHeader.parentIds );
```

Comment:
Lets also make sure `parentIds` is set to a value regardless of whether it exists in the extension - even if that's an empty array or "null".

Reply:
I copied `classIds` as a default values for `parentIds`, since an instance whose parent is itself is considered as without parent (as mentioned [here](https://github.com/CesiumGS/3d-tiles/blob/87218a7c2030890b5200cf16cb466dbbd1480a20/extensions/3DTILES_batch_table_hierarchy/README.md?plain=1#L194)).

I get that we want it done to keep consistency in class properties, but be aware that this could cause issue in a specific case. Indeed, if user has data with a `parentCounts` different from an array of `1` and has no defined `parentIds`, this will create unwanted inheritance between instances that should not inherit any property. I get that this is a weird situation (having a `parentCounts` without `parentIds`) but that's not something prohibited by the specification.

Reply:
I think it's okay to set `parentIds` to `null`, and that sounds like the better option given your explanation - I just wanted it to be set to _something_ in the general case so it's more clear what's going on when debugging.

Reply:
Good idea setting `parentIds` to `null`. I changed it.